pallets/dapp-staking/src/lib.rs

```diff
@@ -2691,6 +2692,47 @@ pub mod pallet {
 
             Ok(())
         }
+
+        /// ### Invariants of `EraInfo`
+        ///
+        /// 1. StakerInfo total voting stake == CurrentEraInfo.next_stake_amount (if same period)
+        /// 2. Current voting stake ≤ Next voting stake (if same period) (not equal due to possible moves)
+        #[cfg(any(feature = "try-runtime", test))]
+        pub fn try_state_era_info() -> Result<(), sp_runtime::TryRuntimeError> {
+            let protocol_state = ActiveProtocolState::<T>::get();
+            let current_period = protocol_state.period_number();
+            let era_info = CurrentEraInfo::<T>::get();
+
+            let current_voting = era_info.staked_amount(Subperiod::Voting);
+            let next_voting = era_info.staked_amount_next_era(Subperiod::Voting);
+
+            // Yield voting stake amounts in [`StakerInfo`] for the current period
+            let voting_total_staked: Balance = StakerInfo::<T>::iter()
+                .filter(|(_, _, info)| info.period_number() == current_period)
+                .map(|(_, _, info)| info.staked_amount(Subperiod::Voting))
+                .sum();
+
+            let era_info_next_period = era_info.next_stake_amount.period;
+            let era_info_current_period = era_info.current_stake_amount.period;
```

Comment:
I'm a bit confused with this test - won't these two numbers be the same?

How can it happen that there are two different period numbers here?

Reply:
I thought that for the last era of a period, the next stake might have already switched to the next period. After checking era_and_period_handler, the next and current stake periods are always in sync, so I’m removing the check from the if.